src/wp-includes/rest-api.php

```diff
@@ -1366,6 +1366,14 @@ function rest_validate_value_from_schema( $value, $args, $param = '' ) {
 				)
 			);
 		}
+
+		if ( isset( $args['pattern'] ) ) {
+			$pattern = str_replace( '#', '\\#', $args['pattern'] );
```

Comment:
```suggestion
			$pattern = preg_quote( $args['pattern'], '#' );
```

Reply:
Ah totally missed the related trac conversation you're right this would be incorrect.